### PR TITLE
Jenkins MVP

### PR DESCRIPTION
**Pull request by sirspinach**
This series adds a spec and defines a Dockerfile for a Quilt Jenkins instance. Every hour, the Jenkins instance runs a slightly modified version of quilt-tester and posts results to the #quilt-jenkins Slack channel.

### CODE REVIEW

**Comment by sirspinach** [comment]
I should add some documentation for building the Jenkins docker image.

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,5 @@
 From quilt/ovs
 Maintainer Ethan J. Jackson
 
-Copy ./buildinfo /buildinfo
```
Comment (sirspinach) [inline Dockerfile]: should also stop generating buildinfo elsewhere in Makefile

File: quilt-tester/io.go [hunk 2]
```diff
@@ -50,18 +49,9 @@ func newLogger() (logger, error) {
 	if err := os.MkdirAll(failedDir, 0755); err != nil {
 		return logger{}, err
 	}
-	if err := exec.Command("cp", "/buildinfo", buildinfoPath).Run(); err != nil {
-		logrus.WithError(err).Error("Failed to copy build info.")
-	}
```
Comment (sirspinach) [inline quilt-tester/io.go]: This buildinfo change belongs in the previous commit.

File: quilt-tester/io.go [hunk 3]
```diff
@@ -35,11 +36,9 @@ func (l logger) testLogger(passed bool, testName string) fileLogger {
 // Create a new logger that will log in the proper directory.
 // Also initializes all necessary directories and files.
 func newLogger() (logger, error) {
-	webDir := filepath.Join(webRoot, time.Now().Format("02-01-2006_15h04m05s"))
-	passedDir := filepath.Join(webDir, "passed")
-	failedDir := filepath.Join(webDir, "failed")
-	logDir := filepath.Join(webDir, "log")
-	buildinfoPath := filepath.Join(webDir, "buildinfo")
```
Comment (sirspinach) [inline quilt-tester/io.go]: same here.

**Comment by ejj** [comment]
@sirspinach what's the status of this?  Is it ready?

**Comment by sirspinach** [comment]
@ejj It runs properly and is ready for review, if that's what you are asking!

**Comment by kayousterhout** [comment]
@sirspinach what's the status with the comments you've left?

**Comment by sirspinach** [comment]
@kayousterhout Just made those changes now, sorry for the confusion.

**Comment by kayousterhout** [comment]
Ok thanks @sirspinach! FYI I won't get to this until tomorrow afternoon.

File: quilt-tester/quilt-tester.go [hunk 4]
```diff
@@ -17,33 +17,31 @@ import (
 )
 
 const (
-	testRoot           = "/tests"
-	quiltPath          = "/.quilt"
+	quiltPath          = "~/.quilt"
```
Comment (kayousterhout) [inline quilt-tester/quilt-tester.go]: Why is the spacing of the "=" here different than below?

File: quilt-tester/config/jenkins_home/plugins.txt [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+pipeline-model-definition:latest
+pipeline-rest-api:latest
+pipeline-stage-step:latest
+pipeline-stage-view:latest
+plain-credentials:latest
+resource-disposer:latest
+run-condition:latest
+scm-api:latest
+script-security:latest
+slack:latest
```
Comment (kayousterhout) [inline quilt-tester/config/jenkins_home/plugins.txt]: Where does this get configured?

**Comment by ejj** [comment]
@sirspinach what's the status of this?

**Comment by sirspinach** [comment]
@ejj: working on floating ip/slack display today

**Comment by kklin** [comment]
Should I pick this up?

**Comment by sirspinach** [comment]
yeah, maybe you should. I can answer any questions you have on Slack

**Comment by kklin** [comment]
Will re-open when my changes are ready.

**Pull request closed without merging**